network/oauth_client.go

```diff
@@ -64,6 +64,7 @@ func (oc OAuthClient) Do(request *http.Request) (*http.Response, error) {
 	}
 
 	client := oc.oauthConfig.Client(oc.context, token)
+	client.Timeout =  oc.timeout
```

Comment:
This client.Timeout actually should be the same as the one that was on the `http.Client` above. Can we please add it back to that struct so that all of the relevant options at construction are grouped together?